### ISSUE TITLE
Bad request: Empty update request when  points.len() < chunk_size missing points bug

### ISSUE THREAD

**Issue by lijingrs** [qdrant/rust-client#174]
pub async fn upsert_points_chunked(
        &self,
        request: impl Into<UpsertPoints>,
        chunk_size: usize,
    ) -> QdrantResult<PointsOperationResponse> {
        let mut request = request.into();
        let points = std::mem::take(&mut request.points);

        if points.len() < chunk_size {
            return self.upsert_points(request).await;
        }

        let request = &request;
        let points = &points;

        self.with_points_client(|mut points_api| async move {
            let mut resp = PointsOperationResponse {
                result: None,
                time: 0.0,
            };

            for chunk in points.clone().chunks(chunk_size) {
                let mut chunked_request = request.clone();
                chunked_request.points = chunk.to_vec();

                let PointsOperationResponse { result, time } =
                    points_api.upsert(chunked_request).await?.into_inner();

                resp.result = result;
                resp.time += time;
            }

            Ok(resp)
        })
        .await
    }

**Comment by timvisee** [comment]
What is the exact problem here?

Are you saying that this `_chunked` function doesn't work properly if the given set of points is smaller than the chunk size? If so, could you provide an reproducible example for this.

**Comment by lijingrs** [comment]
> What is the exact problem here?
> 
> Are you saying that this `_chunked` function doesn't work properly if the given set of points is smaller than the chunk size? If so, could you provide an reproducible example for this.

let upsert_points = UpsertPointsBuilder::new(
                        text_collection_name.to_string(),
                        text_point_structs.clone(),
                    );
 let _text_result = qdrant_client.upsert_points_chunked(upsert_points,300).await;
Executing this code will throw an exception when the length of upsert_points is less than 300.

**Comment by timvisee** [comment]
You're totally right. Thank you for diving into it. I'll make sure to release a patch shortly.
